src/commands/General/mergepdf.ts

```diff
@@ -0,0 +1,38 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import type {
+ Args, CommandOptions,
+} from '@sapphire/framework';
+import PdfMerger from 'pdf-merger-js';
+import { mergePDF as config } from '@/config/commands/general';
+import HorizonCommand from '@/structures/commands/HorizonCommand';
+import type { GuildMessage } from '@/types';
+
+@ApplyOptions<CommandOptions>(config.options)
+export default class PdfMergeCommand extends HorizonCommand {
+  public async run(message: GuildMessage, args: Args): Promise<void> {
+    // Argument dans l'ordre croissant
+    const result = await args.repeatResult('message');
+
+    if (!result.success) {
+        await message.channel.send(config.messages.noPdfFound);
+        return;
+    }
+    const merger = new PdfMerger();
+
+    for (const msg of result.value) {
+        for (const fichier of msg.attachments.values()) {
+            if (fichier.name.endsWith('.pdf'))
+                merger.add(fichier.proxyURL);
```

Comment:
```suggestion
        for (const file of msg.attachments.values()) {
            if (file.name.endsWith('.pdf'))
                merger.add(file.url);
```
Renom de `fichier` en `file`, et utilisation de `file.url` plutôt que `.proxyURL`, car autant utiliser son lien direct plutôt que le lien de son proxy :)

Et d'ailleurs merger.add attend un fichier (donc un Buffer ou un chemin vers un fichier local). Donc il faut télécharger ces PDFs pour pouvoir les fusionner.
Pour cela tu peux utiliser le module [`env-path`](https://www.npmjs.com/package/env-paths) pour pouvoir récupérer le chemin vers `/tmp/` selon l'OS.
```js
import envPaths from 'env-paths';

// Path to "tmp" folder, depending on the OS
const tmpFolder = envPaths('horizonbot').temp;

// le reste de la classe
```
comme ca tu créés une variable global "tmpFolder", en dehors de la classe, que tu peux accéder.
Ensuite il faut télécharger les fichiers, les mettre dedans, les donner à `merger.add`, puis à la fin les supprimer :)